exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -494,6 +490,39 @@ func (b *Bitfinex) handleWSEvent(respRaw []byte) error {
 	return nil
 }
 
+// handleWSSubscribed parses a subscription response and registers the chanID key immediately, before updating subscribeToChan via IncomingWithData chan
+// handleWS* happens sequentially, so by rekeying on chanID immediately we ensure the first message is not dropped
+func (b *Bitfinex) handleWSSubscribed(respRaw []byte) error {
+	subID, err := jsonparser.GetUnsafeString(respRaw, "subId")
+	if err != nil {
+		return fmt.Errorf("%w 'subId': %w from message: %s", errParsingWSField, err, respRaw)
+	}
+
+	c := b.Websocket.GetSubscription(subID)
+	if c == nil {
+		return fmt.Errorf("%w: %w subID: %s", stream.ErrSubscriptionFailure, stream.ErrSubscriptionNotFound, subID)
+	}
+
+	chanID, err := jsonparser.GetInt(respRaw, "chanId")
+	if err != nil {
+		return fmt.Errorf("%w: %w 'chanId': %w; Channel: %s Pair: %s", stream.ErrSubscriptionFailure, errParsingWSField, err, c.Channel, c.Currency)
+	}
+
+	// Note chanID int type avoids conflicts with subID key
```

Comment:
The types are different. So we know that a subID key can't clash with a chanID key.
Arose when I was concerned about a clash, and then realised it's implicitly impossible, but wanted to make that explicit in a comment.

Changed to:
`Note: chanID's int type avoids conflicts with the string type subID key because of the type difference`